src/components/icon/icon-registry.ts

```diff
@@ -0,0 +1,239 @@
+    return this._fontClassNamesByAlias.get(alias) || alias;
+  }
+
+  private _fetchUrl(url: string): Observable<string> {
+    // Store in-progress fetches to avoid sending a duplicate request for a URL when there is
+    // already a request in progress for that URL. It's necessary to call share() on the
+    // Observable returned by http.get() so that multiple subscribers don't cause multiple XHRs.
+    if (this._inProgressUrlFetches.has(url)) {
+      return this._inProgressUrlFetches.get(url);
+    }
+    const req = this._http.get(url)
+        .map((response) => response.text())
+        .finally(() => {
+          this._inProgressUrlFetches.delete(url);
+        })
+        .share();
+    this._inProgressUrlFetches.set(url, req);
+    return req;
+  }
+
+  private _loadIconFromConfig(config: IconConfig): Observable<SVGElement> {
+    return this._fetchUrl(config.url)
+        .map(svgText => this._createSvgElementForSingleIcon(svgText, config));
+  }
+
+  private _loadIconSetFromConfig(config: IconConfig): Observable<SVGElement> {
+    return this._fetchUrl(config.url)
+        .map((svgText) => this._svgElementFromString(svgText));
+  }
+
+  private _createSvgElementForSingleIcon(responseText: string, config: IconConfig): SVGElement {
+    const svg = this._svgElementFromString(responseText);
+    this._setSvgAttributes(svg, config);
+    return svg;
+  }
+
+  private _setSvgAttributes(svg: SVGElement, config: IconConfig) {
+    if (!svg.getAttribute('xmlns')) {
+      svg.setAttribute('xmlns', 'http://www.w3.org/2000/svg');
+    }
+    svg.setAttribute('fit', '');
+    svg.setAttribute('height', '100%');
+    svg.setAttribute('width', '100%');
+    svg.setAttribute('preserveAspectRatio', 'xMidYMid meet');
+    svg.setAttribute('focusable', 'false'); // Disable IE11 default behavior to make SVGs focusable.
+  }
+
+  private _extractSvgIconFromSet(
+      iconSet: SVGElement, iconName: string, config: IconConfig): SVGElement {
+    const iconNode = iconSet.querySelector('#' + iconName);
+    if (!iconNode) {
+      return null;
+    }
+    // createElement('SVG') doesn't work as expected; the DOM ends up with
+    // the correct nodes, but the SVG content doesn't render. Instead we
+    // have to create an empty SVG node using innerHTML and append its content.
+    // http://stackoverflow.com/questions/23003278/svg-innerhtml-in-firefox-can-not-display
+    const svg = this._svgElementFromString('<svg></svg>');
+    // Clone the node so we don't remove it from the parent icon set element.
+    svg.appendChild(iconNode.cloneNode(true));
+    this._setSvgAttributes(svg, config);
+    return svg;
+  }
+
+  private _svgElementFromString(str: string): SVGElement {
+    // TODO: Is there a better way than innerHTML? Renderer doesn't appear to have a method for
+    // creating an element from an HTML string.
+    const div = document.createElement('DIV');
+    div.innerHTML = str;
+    const svg = <SVGElement>div.querySelector('svg');
```

Comment:
This prompted me to file an issue on the TypeScript repo, since it _could_ statically infer what the type is.
https://github.com/Microsoft/TypeScript/issues/8114